src/makeCalculator.js

```diff
@@ -4,7 +4,23 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    operate(action, number) {
+      this.result = action(this.result, number);
+
+      return this;
+    },
+    add: (a, b) => a + b,
+    subtract: (a, b) => a - b,
+    multiply: (a, b) => a * b,
+    divide: (a, b) => a / b,
```

Comment:
Arrow functions should not be used for methods that will be using `this` from their containing object, because arrow functions do not have their own `this` context; they inherit `this` from the parent scope at the time they are defined. In this case, since the methods are using `this.result` in the `operate` method, they should be defined using the shorthand method syntax without the arrow function.